docs/features/ValueTuples.cs

```diff
@@ -0,0 +1,537 @@
+   }
+}
+
+namespace System
+{
+    public static class ValueTuple
+    {
+        public static ValueTuple<T1> Create<T1>(T1 item1) =>
+            new ValueTuple<T1>(item1);
+
+        public static ValueTuple<T1, T2> Create<T1, T2>(T1 item1, T2 item2) =>
+            new ValueTuple<T1, T2>(item1, item2);
+
+        public static ValueTuple<T1, T2, T3> Create<T1, T2, T3>(T1 item1, T2 item2, T3 item3) =>
+            new ValueTuple<T1, T2, T3>(item1, item2, item3);
+
+        public static ValueTuple<T1, T2, T3, T4> Create<T1, T2, T3, T4>(T1 item1, T2 item2, T3 item3, T4 item4) =>
+            new ValueTuple<T1, T2, T3, T4>(item1, item2, item3, item4);
+
+        public static ValueTuple<T1, T2, T3, T4, T5> Create<T1, T2, T3, T4, T5>(T1 item1, T2 item2, T3 item3, T4 item4, T5 item5) =>
+            new ValueTuple<T1, T2, T3, T4, T5>(item1, item2, item3, item4, item5);
+
+        public static ValueTuple<T1, T2, T3, T4, T5, T6> Create<T1, T2, T3, T4, T5, T6>(T1 item1, T2 item2, T3 item3, T4 item4, T5 item5, T6 item6) =>
+            new ValueTuple<T1, T2, T3, T4, T5, T6>(item1, item2, item3, item4, item5, item6);
+
+        public static ValueTuple<T1, T2, T3, T4, T5, T6, T7> Create<T1, T2, T3, T4, T5, T6, T7>(T1 item1, T2 item2, T3 item3, T4 item4, T5 item5, T6 item6, T7 item7) =>
+            new ValueTuple<T1, T2, T3, T4, T5, T6, T7>(item1, item2, item3, item4, item5, item6, item7);
+
+        public static ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest> Create<T1, T2, T3, T4, T5, T6, T7, TRest>(T1 item1, T2 item2, T3 item3, T4 item4, T5 item5, T6 item6, T7 item7, TRest rest)
+            where TRest : IEquatable<TRest>
+        {
+            return new ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>(item1, item2, item3, item4, item5, item6, item7, rest);
+        }
+    }
+
+    // struct with one value
+    public struct ValueTuple<T1> : IEquatable<ValueTuple<T1>>
+    {
+        private static readonly EqualityComparer<T1> s_comparer1 = EqualityComparer<T1>.Default;
+
+        public T1 Item1;
+
+        public ValueTuple(T1 item1)
+        {
+            this.Item1 = item1;
+        }
+
+        public bool Equals(ValueTuple<T1> other)
+        {
+            return s_comparer1.Equals(this.Item1, other.Item1);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is ValueTuple<T1>)
+            {
+                var other = (ValueTuple<T1>)obj;
+                return this.Equals(other);
+            }
+
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return s_comparer1.GetHashCode(Item1);
+        }
+
+        public static bool operator ==(ValueTuple<T1> left, ValueTuple<T1> right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(ValueTuple<T1> left, ValueTuple<T1> right)
+        {
+            return !left.Equals(right);
+        }
+    }
```

Comment:
I assume there will be additional members like "ToString", ITuple implementation, perhaps "is" operator if needed...

As a minimal implementation this seems good enough.